config.go

```diff
@@ -108,7 +110,9 @@ func (c *ReaderConfig) Apply(options ...ReaderOption) {
 
 // ConfigureReader applies configuration options from c to config.
 func (c *ReaderConfig) ConfigureReader(config *ReaderConfig) {
```

Comment:
This feels like sort of a weird API now to be honest. Maybe

```
func (config *ReaderConfig) Clone() *ReaderConfig {
}
```

would be a better API?

Reply:
I'm not sure if you mean the `ConfigureReader` API is weird or the implementation of the method. Could you give a bit more details about what we should change?

Reply:
To give a bit more color to this, `ConfigureReader` is meant to implement the [`ReaderOption`](https://pkg.go.dev/github.com/segmentio/parquet-go#ReaderOption) interface.

Reply:
Sorry just the whole semantics of already having a ReaderConfig, and then passing a ReaderConfig (which already has some configuration) to that ReaderConfig to be modified, just seems weird.

What about, I don't know, 

```
// Coalesce modifies the child ReaderConfig using the schema present in parent
func (parent *ReaderConfig) Coalesce (child *ReaderConfig) {
}
```

or 

```
// ConfigFromSchema coalesces the schema in parent and the schema in child, returning a ReaderConfig
// with the coalesced schema
func (parent *ReaderConfig) ConfigFromSchema (child *Schema) *ReaderConfig {
}
```

or something?